script.js

```diff
@@ -1,66 +1,151 @@
+    }
+
+    markDone(){
+        if(!this.isDone()){
+            this.taskInput.classList.add(DONE_TASK_TEXT_CLASS);
+        }
+    }
+
+    unMarkDone(){
+        if(this.isDone()){
+            this.taskInput.classList.remove(DONE_TASK_TEXT_CLASS);
+        }
+    }
+
+    changeTaskDoneMark() {
+        if (this.isDone()) {
+            this.unMarkDone()
+        } else {
+            this.markDone()
+            moveToBottom(this);
+        }
+    }
+
+    isDone(){
+        return this.taskInput.classList.contains(DONE_TASK_TEXT_CLASS);
+    }
+
+    deleteTask(){
+        content.removeChild(this.div);
+        this.div.remove();
+    }
+
+    saveToLocalStorage(){}
 }
 
-function addTaskDeleteButton(task) {
-    let deleteButton = document.createElement("button");
-    deleteButton.textContent = "X";
-    deleteButton.classList.add("delete-button");
-    deleteButton.onclick = () => { deleteTask(task) };
-    task.appendChild(deleteButton);
+function addEmptyTask() {
+    let task = new Task();
+    if (!content.hasChildNodes()){
+        content.appendChild(task.div);
+    } else {
+        content.insertBefore(task.div, content.firstChild);
+    }
+    
 }
 
-function addTaskFinishedButton(task, taskInput) {
-    let finishedButton = document.createElement("button");
-    finishedButton.textContent = '\u{02713}';
-    finishedButton.classList.add("finished-button");
-    finishedButton.onclick = () => { chageTaskDoneMark(taskInput) };
-    task.appendChild(finishedButton);
+function moveToBottom(task){
+    content.removeChild(task.div);
+    content.appendChild(task.div);
 }
 
-function createEmptyTask() {
-    let newTask = createRawTask();
-    taskInput = addTaskInputBar(newTask)
-    addTaskDeleteButton(newTask)
-    addTaskFinishedButton(newTask, taskInput)
-    return newTask;
+function removeAllDoneTasks(){
+    childrenCopy = [... content.children]
+    for (child of childrenCopy){
+        if (child.task.isDone()){
+            child.task.deleteTask();
+        }
+    }
 }
 
-function deleteTask(task) {
-    content.removeChild(task);
-    task.remove();
+function removeAllTasks(){
+    childrenCopy = [... content.children]
+    for (child of childrenCopy){
+        child.task.deleteTask();
+    }
 }
 
-function chageTaskDoneMark(taskInput) {
-    if (taskInput.classList.contains(DONE_TASK_TEXT_CLASS)) {
-        taskInput.classList.remove(DONE_TASK_TEXT_CLASS);
-    } else {
-        taskInput.classList.add(DONE_TASK_TEXT_CLASS);
+function loadFromLocalStorage(){
+    states = JSON.parse(localStorage.getItem("currentState"));
+    if (states && states.length > 0) {
+        for (state of states) {
+            let task = new Task();
+            task.setState(state);
+            content.appendChild(task.div);
+        }
     }
 }
 
+function saveLocalStorage(){
+    states = Array.from(content.children).map((child)=>child.task.getState());
+    localStorage.setItem("currentState", JSON.stringify(states));
```

Comment:
the key on localStorage have to be a constant

Reply:
done